pkg/model/config/config.go

```diff
@@ -78,39 +81,105 @@ func (c Config) IsV3() bool {
 	return c.Version == Version3Alpha
 }
 
-// HasResource returns true if API resource is already tracked
-func (c Config) HasResource(target GVK) bool {
-	// Return true if the target resource is found in the tracked resources
+type unsupportedFieldError struct {
+	fieldName string
+	version   string
+}
+
+func (e unsupportedFieldError) Error() string {
+	return fmt.Sprintf("`%s` field found for %s", e.fieldName, e.version)
+}
+
+func (c Config) errorForField(fieldName string) error {
+	return unsupportedFieldError{fieldName: fieldName, version: c.Version}
+}
+
+// validate checks if a Config is valid or it has some fields that shouldn't be present for that version
+func (c Config) validate() error {
```

Comment:
Wdyt about adding a name her is that is more meaningful? I mean, make clear its goal is to ensure that config has not an attribute that is invalid for v2? 

Also, I do not like this approach, we are adding business logic in the persistent layer which shows hurts a few concepts. The validation of what is or not valid for tthe plugin should be in the plugin instead of it.

Reply:
I plan to turn `config.Config` into an interface and maek an implementation for each project version which will remove all the conditionals about some specific version in a following PR.

Reply:
Then, we can change the name and etc. However,  until there, wdyt about we try to keep the current code more understandable and intuitive?